### PR TITLE
Exports metric for WAL write errors

### PR DESCRIPTION
**Pull request by jpmcb**
Related to https://github.com/prometheus/prometheus/issues/6577

Includes a metric for WAL write errors

### CODE REVIEW

File: tsdb/head.go [hunk 1]
```diff
@@ -178,6 +179,10 @@ func newHeadMetrics(h *Head, r prometheus.Registerer) *headMetrics {
 		Name: "prometheus_tsdb_wal_corruptions_total",
 		Help: "Total number of WAL corruptions.",
 	})
+	m.walWritesFailed = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "prometheus_tsdb_wal_writes_failed",
```
Comment (cstyan) [inline tsdb/head.go]: counters should end with `_total`

Comment (jpmcb) [reply tsdb/head.go]: Ah yes, thanks for that! Updated!!

**Comment by brian-brazil** [comment]
:+1:

**Comment by cstyan** [comment]
would be easier to move the metric to `wal.go`: https://github.com/prometheus/prometheus/blob/master/tsdb/wal/wal.go#L183-L190

**Comment by jpmcb** [comment]
Thanks for the feedback! Reading through `wal.go`, looks like whenever the page gets flushed, we do a file write. That error block seemed like the best place for the metric, but let me know if I missed anything else!

**Comment by cstyan** [comment]
I think the error should be tracked in `Log(...)`, that's the way external consumers can write to the log and that function can error for reasons other than just writing to disk.

**Comment by jpmcb** [comment]
Oh I see - `head.go` and other outside consumers will call `wal.Log(rec)`. I've made the change, thanks so much for the help on this one and bringing me up to speed!

**Pull request merged**